src/ICeWall/src/ICeWall.tcl

```diff
@@ -3561,10 +3713,914 @@ namespace eval ICeWall {
+      }
+    }
+    return [check_location $location]
+  }
+
+  proc check_bump {bump} {
+    variable footprint
+
+    if {[dict exists $footprint type]} {
+      if {[dict get $footprint type] != "Flipchip"} {
+        utl::error PAD 106 "Specification of bumps is only allowed for Flipchip padring layouts"
+      }
+    }
+
+    if {[catch {check_rowcol $bump} msg]} {
+      utl::error PAD 107 "Bump value specified incorrectly, $msg"
+    }
+
+    return $bump
+  }
+
+  proc check_inst_name {padcell} {
+    variable block
+    variable footprint
+
+    if {[dict exists $padcell inst_name]} {
+      set inst_name [dict get $padcell inst_name]
+    } else {
+      if {[dict exists $padcell use_signal_name] && ([is_power_net [dict get $padcell use_signal_name]] || [is_ground_net [dict get $padcell use_signal_name]])} {
+        set inst_name [format [dict get $footprint pad_inst_name] [dict get $padcell name]]
+      } elseif {[set inst_name [check_inst_name_versus_signal $padcell]] == "NULL"} {
+        if {[dict exists $footprint pad_inst_name]} {
+          if {[dict exists $padcell use_signal_name]} {
+            set inst_name [format [dict get $footprint pad_inst_name] [dict get $padcell use_signal_name]]
+          } else {
+            set inst_name [format [dict get $footprint pad_inst_name] [dict get $padcell name]]
+          }
+        } else {
+          if {[dict exists $padcell use_signal_name]} {
+            set inst_name [dict get $padcell use_signal_name]
+          } else {
+            set inst_name [dict get $padcell name]
+          }
+        }
+      }
+    }
+
+    if {[set inst [$block findInst $inst_name]] == "NULL"} {
+      set cell [get_padcell_cell_name $padcell]
+      if {[is_padcell_physical_only $padcell]} {
+        # debug "Create physical_only cell $cell with name $inst_name for padcell $padcell"
+        # debug "master: [get_cell_master $cell]"
+        set inst [odb::dbInst_create $block [get_cell_master $cell] $inst_name]
+        dict set padcell inst $inst
+        # debug "done"
+      } elseif {[is_padcell_control $padcell]} {
+        # debug "Create control cell $cell with name $inst_name for padcell $padcell"
+        set inst [odb::dbInst_create $block [get_cell_master $cell] $inst_name]
+        dict set padcell inst $inst
+      } elseif {[is_footprint_create_padcells]} {
+        # debug "Create cell $cell with name $inst_name for padcell $padcell"
+        set inst [odb::dbInst_create $block [get_cell_master $cell] $inst_name]
+        dict set padcell inst $inst
+      } elseif {[is_padcell_power $padcell] || [is_padcell_ground $padcell]} {
+        # debug "Create power/ground $cell with name $inst_name for padcell $padcell"
+        set inst [odb::dbInst_create $block [get_cell_master $cell] $inst_name]
+        dict set padcell inst $inst
+      } else {
+        debug $padcell
+        utl::error PAD 122 "Cannot find an instance with name \"$inst_name\""
```

Comment:
@Colin-Holehouse there appears to be an issue when calling `add_pad` with a pad that is not in the netlist.
I'm trying to add power and ground pads and it looks like `is_padcell_power`  and `is_padcell_ground` are returning 0. This looks like `get_padcell_assigned_name` isn't returning the signal that the pad is connected to, but instread is just returning the dict that was passed in.

Reply:
I will ensure this will work as you're expecting in the next update to this PR, and power/ground cells that are not in the input design can be added to the design this way.